### PR TITLE
Feat/fluid input as webcomponent

### PR DESCRIPTION
**Pull request by peter-affenzeller**
### <strong>Pull Request</strong>

FluidInput web-component

#### Type of PR

Feature (non-breaking change which adds functionality)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by ffriedl89** [comment]
I've tried to use the input and I think it does a bit too much right now. I'd argue that the hint & label & potential validation messages are not bound to the input alone but rather to any form control. 
Having a `fluid-input` that maybe just proxies the HTMLInputElement specific properties seems better after i had a chance to try it out. The markup for the hint, errors and label should in my opinion be moved to a form-field component.

E.g. 
```
<fluid-form-field name="filterby">
    <label>Filter by</label>
    <fluid-input type="text"></dt-input>
</fluid-form-field>
```

**Review by heartdisease — APPROVED** [review]

File: .vscode/extensions.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
     "esbenp.prettier-vscode",
     "ms-vscode.vscode-typescript-tslint-plugin",
     "angular.ng-template",
-    "hex-ci.stylelint-plus"
+    "hex-ci.stylelint-plus",
+    "runem.lit-plugin"
```
[the remark this answers is not in the document]
Comment (peter-affenzeller) [reply .vscode/extensions.json:8]: for syntax-highlighting:

File: libs/fluid-elements/input/src/lib/input.spec.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    }
+    document.body.innerHTML =
+      '<fluid-input hint="FluidInput hint."><label slot="label">Fluid Input Label</label><input type="text" placeholder="Fluid Input" /></fluid-input>';
+    fixture = document.querySelector<FluidInput>('fluid-input')!;
+  });
+
+  it('should create the input', async () => {
+    expect(fixture).not.toBe(null);
+  });
+});
```
Comment (lukasholzer) [inline libs/fluid-elements/input/src/lib/input.spec.ts:34]: please add some more test cases :D or create follow up issue for it

File: libs/fluid-elements/input/src/lib/input.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  private _handleSlotChange({ target }: { target: HTMLSlotElement }): void {
+    const slottedElements = target.assignedElements({
+      flatten: true,
+    });
+
+    // Only accept one element per slot
+    if (slottedElements.length > 1) {
+      throw new Error(
+        `The fluid-input only takes one input element in the default slot.`,
+      );
```
Comment (lukasholzer) [inline libs/fluid-elements/input/src/lib/input.ts:115]: @ffriedl89 do we really want to throw an error or is logging enough?

**Review by lukasholzer — DISMISSED** [review]

File: libs/fluid-elements/input/src/lib/input.ts (hunk 3 above)
Comment (ffriedl89) [reply libs/fluid-elements/input/src/lib/input.ts:115]: I think with a changed API where the form field is extracted we can add a warning/error log. Throwing an error is usually not the go to i'd say. For now I wouldn't bother.

**Review by heartdisease — APPROVED** [review]

**Review by lukasholzer — APPROVED** [review]

**Review by bardosmisi — APPROVED** [review]
I think the input component should be added to the fluid-dev app in order to present what it can do and how does it look like. The other, already done fluid components are also in there (button, checkbox, tab, switch).

Also I guess the component should have a name and a value property in order to make it serializable, but that can be done in the future as well.

Otherwise LGTM 🦛 !

**Comment by lukasholzer** [comment]
Manual cherry-picked into `8.x` and `8.0.x`

**Comment by peter-affenzeller** [comment]
thx @bardosmisi, I added this to the follow-up issue (#1596)

**Pull request merged**